Popup window

Maxwell011 opened the pull request:
Implemented the following interactions:

When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
When the user clicks (or taps) the close (X) button, the popup disappears.
In order to associate each project with the popup window details, I refactor the project section:

- You need to use a JavaScript array to store all of the information for all projects.
- For each project you need to store the following pieces of data in a JavaScript object, at a minimum: name, description,  featured image, technologies, link to live version, and link to the source.
I updated the main page so the projects section is created dynamically using the information stored in that JavaScript object. 
After that, I implement the popup window.
I implement popups for both mobile and desktop screen sizes:
- Mobile popup
- Desktop popup

Kindly Review 🙏

Maxwell011 committed:
Added modal element

Maxwell011 committed:
Added modal styles

Maxwell011 committed:
Added popup js file and popup elements

Maxwell011 committed:
Added more image to images folder

johnkioko committed:
Added onclick function to the mobile popup

johnkioko committed:
Add onclick function

Maxwell011 committed:
Added functionality to the modal

Maxwell011 committed:
Added styles to for the modal

thecodechaser commented on index.html:
```diff
@@ -104,6 +104,12 @@ <h1 class="heading">
+        class="modal modal-dialog modal-dialog-centered modal-dialog-scrollable"
+      ></section>
+
       <!-- Works/projects section-->
       <section class="grid-container">
         <h2 class="heading-container heading-one" id="portfolio">
```
- [ ] The works section should be created dynamically using the information stored in that JavaScript object as requested in the project requirements. Please update the works section and make sure that all of the HTML in that section is created when the page loads. `(You should delete the works section from the main page and create it in the javascript file dynamically).`

![image](https://user-images.githubusercontent.com/83803414/189145703-eb9984b4-b63f-48c9-b5ff-8633d0dddb13.png)

thecodechaser commented on popsup.js:
```diff
@@ -0,0 +1,181 @@
+  <ul class="tags" id="mobile-ul">
+  ${project.technologiesMobile.map((ele) => `<li>${ele}</>`)}
+  </ul>
+  <p id="description">${project.description}</p>
+  <div class="modal-footer">
```
- [ ] On the desktop screen the project description is overflowing from the model, Please fix this issue and make sure that the description is displayed inside the model and your design matches with the [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A468) design.

### Your design
![image](https://user-images.githubusercontent.com/83803414/189147559-a1e601f2-8e13-4617-982a-5e03181ca3dc.png)

### Figma design
![image](https://user-images.githubusercontent.com/83803414/189147380-0da2a78b-b6dd-4d43-a97e-0c9e3ecf561d.png)

thecodechaser commented on popsup.js:
```diff
@@ -0,0 +1,181 @@
+  </ul>
+  <p id="description">${project.description}</p>
+  <div class="modal-footer">
+  <a href="${project.liveLink}">
+    <button type="submit" class="project-view-btn">
+      <span> See Live</span>
+      <img
+        src="${project.liveIcon}"
+        alt="See Live Project"
+      /></button
+  ></a>
+  <a href="${project.sourceLink}">
+    <button type="submit" class="project-view-btn">
+      <span>See Source</span>
+      <img
+        src="${project.githubIcon}"
+        alt="See Source Code"
+      /></button
+  ></a>
```
- [ ] Please remove the black border from the buttons and make sure the button's texts have equal padding from all sides.

### Your design
![image](https://user-images.githubusercontent.com/83803414/189148033-1d7bca37-4b86-4370-ad6b-2763092833cd.png)

### Figma design
![image](https://user-images.githubusercontent.com/83803414/189148076-ebc74355-142b-49e2-8739-af2b7e8bae6a.png)

thecodechaser reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Maxwell011👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️

<img src="https://media2.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Maxwell011 committed:
Added changes reviewer requested

Maxwell011 committed:
Fixed Linter

Maxwell011 commented:
I have added the needed changes asked by the reviewer 👍 

- The Work section has been created Dynamically 
- I have increased the height.
- I have removed the border.

Kindly Review 🙏

victorokoroji reviewed (approved):
Hi @Maxwell011, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅  
- Linters are working properly ✅ 
<img src="https://media3.giphy.com/media/xT0xezQGU5xCDJuCPe/giphy.gif"/>

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
